docs/proposals/diamond.md

```diff
@@ -0,0 +1,39 @@
+# Diamond workflow support.
+
+Diamond workflow allows for a pipeline to scale out the generation of a single bundle. The individual nodes writes to 
+splits of the eventual bundle without any co-ordination between them. At the time of commit any conflicts will be
+dealt with. Conflicts can only occur if the same file in the namespace of the bundle is written to by more than one split.
+
+The diamond workflow consists of the following steps.
+1. Before a diamond starts, Datamon SDK allows for the creation of an unique ID for a diamond.
+   1. This can be a KSUID.
+   2. The client is expected to use the KSUID at each step.
+   3. command: ```datamon bundle split initialize --repo <repo> --message```
+      1. Command will output to STDOUT the id for the diamond.
+   4. backend model: In the vmetadata bucket the following path will be created.
+      ```/splits/repo/<id>/split.yaml```   
+1. Each split in a diamond generates a filelist that is stored in the versioned metadata folder.
+   1. Each split is given a client assigned ID. Example split1,split2,split3.. or 1,2,3.. or a set of UUIDs.
+   2. On retries the split ID should be the same for a given split.
+      1. A node on failure can rerun the same command, as long as the split id is the same
+      the previous filelist contents will be superseded by the ones uploaded later.
+   3. command: ```datamon bundle split add --repo <repo> --split-id <client generated id> --source <path>```
```

Comment:
So i am taking it that `datamon bundle split add` is _actually_ uploading blobs, which will eventually result in a bundle. Correct?

Do we plan to provide the companion command `datamon bundle split remove` to drop/cancel  things? (that would prove quite tricky with our deduplicated blobs?

Reply:
The metadata for split can be removed but the only safe way to remove blobs is via background garbage collection. Not making it part of this PR.